Variadic argument causing panic

### The following program `sample.go` triggers an unexpected result

```go
package main

import "fmt"

type Option func()

func WithOption() Option {
    return func() {}
}

func New(name string, opts ...Option) {
    for _, opt := range opts {
        opt()
    }
}

func main() {
    New("test", []Option{WithOption()}...)
    fmt.Println("test")
}
```


### Expected result

```console
$ go run ./sample.go
test
```


### Got

```console
$ yaegi ./sample.go
/tmp/run.go:12:25: panic
/tmp/run.go:18:26: panic
run: reflect: call of reflect.Value.Call on slice Value
goroutine 1 [running]:
runtime/debug.Stack()
	/usr/lib/go-1.17/src/runtime/debug/stack.go:24 +0x65
github.com/traefik/yaegi/interp.(*Interpreter).Execute.func1()
	/home/xxx/git/yaegi/interp/program.go:131 +0x94
panic({0xd18ee0, 0xc00011c720})
	/usr/lib/go-1.17/src/runtime/panic.go:1038 +0x215
github.com/traefik/yaegi/interp.runCfg.func1()
	/home/xxx/git/yaegi/interp/run.go:193 +0x145
panic({0xd18ee0, 0xc00011c720})
	/usr/lib/go-1.17/src/runtime/panic.go:1038 +0x215
github.com/traefik/yaegi/interp.runCfg.func1()
	/home/xxx/git/yaegi/interp/run.go:193 +0x145
panic({0xd18ee0, 0xc00011c720})
	/usr/lib/go-1.17/src/runtime/panic.go:1038 +0x215
reflect.flag.mustBe(...)
	/usr/lib/go-1.17/src/reflect/value.go:223
reflect.Value.Call({0xcd88a0, 0xc00011c678, 0x4cf1de}, {0x15242c8, 0x7f564c14acc8, 0x8})
	/usr/lib/go-1.17/src/reflect/value.go:337 +0x125
github.com/traefik/yaegi/interp.call.func9(0xc00036c2c0)
	/home/xxx/git/yaegi/interp/run.go:1320 +0x37d
github.com/traefik/yaegi/interp.runCfg(0xc0004937a0, 0xc00036c2c0, 0xc000138a80, 0xcd88a0)
	/home/xxx/git/yaegi/interp/run.go:201 +0x2ac
github.com/traefik/yaegi/interp.call.func9(0xc00036c0b0)
	/home/xxx/git/yaegi/interp/run.go:1436 +0x94e
github.com/traefik/yaegi/interp.runCfg(0xc000494c60, 0xc00036c0b0, 0xc000493d40, 0xc0001e5a80)
	/home/xxx/git/yaegi/interp/run.go:201 +0x2ac
github.com/traefik/yaegi/interp.(*Interpreter).run(0xc00036a000, 0xc000493d40, 0xc00036c000)
	/home/xxx/git/yaegi/interp/run.go:120 +0x385
github.com/traefik/yaegi/interp.(*Interpreter).Execute(0xc00036a000, 0xc000425b30)
	/home/xxx/git/yaegi/interp/program.go:157 +0x24b
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc00036a000, {0xc00010c500, 0xfc}, {0x7ffd7a33e076, 0xc00010c300}, 0xfc)
	/home/xxx/git/yaegi/interp/interp.go:615 +0x5c
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc00036a000, {0x7ffd7a33e076, 0xb})
	/home/xxx/git/yaegi/interp/interp.go:508 +0xab
main.runFile(0x7ffd7a33e076, {0x7ffd7a33e076, 0xb}, 0x0)
	/home/xxx/git/yaegi/cmd/yaegi/run.go:153 +0x8f
main.run({0xc000032050, 0x8001, 0xc00008e6b8})
	/home/xxx/git/yaegi/cmd/yaegi/run.go:116 +0xbec
main.main()
	/home/xxx/git/yaegi/cmd/yaegi/yaegi.go:144 +0x2dd
```


### Yaegi Version

v0.11.2

### Additional Notes

Regression tracked to https://github.com/traefik/yaegi/pull/1362

--- a/_test/issue-1375.go
+++ b/_test/issue-1375.go
@@ -1,0 +1,38 @@
+package main
+
+import "fmt"
+
+type Option func(*Struct)
+
+func WithOption(opt string) Option {
+	return func(s *Struct) {
+		s.opt = opt
+	}
+}
+
+type Struct struct {
+	opt string
+}
+
+func New(opts ...Option) *Struct {
+	s := new(Struct)
+	for _, opt := range opts {
+		opt(s)
+	}
+	return s
+}
+
+func (s *Struct) ShowOption() {
+	fmt.Println(s.opt)
+}
+
+func main() {
+	opts := []Option{
+		WithOption("test"),
+	}
+	s := New(opts...)
+	s.ShowOption()
+}
+
+// Output:
+// test

--- a/interp/run.go
+++ b/interp/run.go
@@ -1235,7 +1235,7 @@ func call(n *node) {
 				values = append(values, genValueInterface(c))
 			case isInterfaceBin(arg):
 				values = append(values, genInterfaceWrapper(c, arg.rtype))
-			case isFuncSrc(arg):
+			case isFuncSrc(arg) && !hasVariadicArgs:
 				values = append(values, genValueNode(c))
 			default:
 				values = append(values, genValue(c))

